config/c-and-cpp.org

```diff
@@ -0,0 +1,11 @@
+* Additional C / C++ components
+
+If you want to use [[https://github.com/MaskRay/ccls][=ccls=]], which supports more language server features than =clangd= like overlays and semantic highlighting,
+
+#+BEGIN_SRC emacs-lisp
+(use-package ccls
+  :after lsp-mode
+  :hook ((c-mode c++-mode) . (lambda ()
+                               (require 'ccls)
```

Comment:
Why do you `(require 'ccls)` when we `use-package` it?

Reply:
Because the package doesn't provide any autoloads based on mode or file extension, and no commands to automatically start the server. There's nothing else possible to add as the hook, and without a hook, `clangd` will start instead.

Reply:
Interesting, back when I used ccls, it would work with the vanilla hook of `((c-mode c++-mode) . lsp)` :man_shrugging: